client/src/panels/panel_declarations/services/services_channels.js

```diff
@@ -0,0 +1,164 @@
+import text from './services.yaml';
+
+import { Service } from '../../../models/services.js';
+
+import {
+  create_text_maker_component,
+  declare_panel,
+  InfographicPanel,
+  NivoResponsiveBar,
+  declarative_charts,
+  util_components,
+} from "../shared.js";
+
+const {
+  Select,
+} = util_components;
+
+const { GraphLegend } = declarative_charts;
+import { Fragment } from 'react';
+
+const { text_maker, TM } = create_text_maker_component(text);
+
+
+
+
+class ServicesChannelsPanel extends React.Component {
+  constructor(){
+    super();
+    this.state = {
+      selected_service_id: "all",
+    };
+  }
+  
+  render(){
+    const {
+      panel_args,
+    } = this.props;
+    const { selected_service_id } = this.state;
+    const colors = infobase_colors();
+
+    const data_keys = ["telephone_enquires",	"website_visits",	"online_applications",	"in_person_applications",	"mail_applications",	"other_channel_applications"];
+
+    const all_services = _.chain({})
+      .mergeWith(..._.map(panel_args.services, serv=>_.pick(serv,data_keys)),_.add)
+      .merge({name: text_maker("all"), id: "all"})
+      .value();
+
+    const selected_service = selected_service_id === "all" ? 
+      all_services :
+      _.find(panel_args.services, serv => serv.id === selected_service_id) || _.first(panel_args.services);
+    
+    const selected_title = selected_service.name;
+
+    const bar_data = _.chain(data_keys)
+      .map(key => ({
+        number: selected_service[key],
+        label: text_maker(`label_${key}`),
+      }) )
+      .each(obj => {obj[text_maker("number_of_enquiries")] = obj.number;})
+      .value();
```

Comment:
Mutation,  😝. We've got a good syntax for declaring objects with variable keys these days, can write this as:
```
const bar_data = _.map(
  data_keys,
  key => ({
    number: selected_service[key],
    [text_maker("number_of_enquiries")]: selected_service[key],
    label: text_maker(`label_${key}`),
  })
);
```